arch/z80/src/z180/z180_mmu.c

```diff
@@ -375,6 +375,33 @@ int up_addrenv_vdata(FAR group_addrenv_t *addrenv, uintptr_t textsize,
   return CONFIG_Z180_COMMON1AREA_VIRTBASE + textsize;
 }
 
+/****************************************************************************
+ * Name: up_addrenv_vheap
+ *
+ * Description:
+ *   Return the heap virtual address associated with the newly created
+ *   address environment.  This function is used by the binary loaders in
+ *   order get an address that can be used to initialize the new task.
+ *
+ * Input Parameters:
+ *   addrenv - The representation of the task address environment previously
+ *      returned by up_addrenv_create.
+ *   vheap - The location to return the virtual address.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int up_addrenv_vheap(FAR const group_addrenv_t *addrenv, FAR void **vheap)
+{
+  /* Not implemented */
+
+  return (ssize_t)-ENOSYS;
```

Comment:
Why we cast to `ssize_t` if API returns `int`?

Reply:
Prototype is wrong, prototype should obviously be size_t or ssize_t. My bad.

Reply:
we can drop the cast

Reply:
should we return the correct value here?